android/jni/tun2socks/tun2socks.c

```diff
@@ -441,22 +453,22 @@ JNIEXPORT jint JNICALL Java_org_uproxy_tun2socks_Tun2SocksJni_runTun2Socks(
     jstring vpnIpAddress,
     jstring vpnNetMask,
     jstring socksServerAddress,
-    jstring dnsServerAddress,
+    jstring dnsResolverAddress,
     jint transparentDNS)
 {
     g_env = env;
 
     const char* vpnIpAddressStr = (*env)->GetStringUTFChars(env, vpnIpAddress, 0);
     const char* vpnNetMaskStr = (*env)->GetStringUTFChars(env, vpnNetMask, 0);
     const char* socksServerAddressStr = (*env)->GetStringUTFChars(env, socksServerAddress, 0);
-    const char* dnsServerAddressStr = (*env)->GetStringUTFChars(env, dnsServerAddress, 0);
+    const char* dnsResolverAddressStr = (*env)->GetStringUTFChars(env, dnsResolverAddress, 0);
 
     init_arguments("uProxy tun2socks");
 
     options.netif_ipaddr = (char*)vpnIpAddressStr;
     options.netif_netmask = (char*)vpnNetMaskStr;
     options.socks_server_addr = (char*)socksServerAddressStr;
-    options.dns_server_address = (char*)dnsServerAddressStr;
+    options.dns_resolver_address = (char*)dnsResolverAddressStr;
```

Comment:
As you know, the UDP relay address is the same as the SOCKS server. I added a comment when we populate `udp_relay_address`. I can explicitly add an option for this in the JNI.